### PR TITLE
fix(sandpack-bundler): Don't use nested pkg.json when exports are present

### PR DESCRIPTION
**Pull request by DeMoorJasper**
## What kind of change does this PR introduce?

We should not use the nested pkg.json's as a reference when the root package.json has an exports field present.

This is based on PR https://github.com/codesandbox/codesandbox-client/pull/7380 this is probably not as spec compliant but at least it doesn't break anything.

We need to investigate this further with a more advanced test suite soonish

### CODE REVIEW

File: packages/sandpack-core/src/resolver/utils/pkg-json.ts [hunk 1]
```diff
@@ -68,5 +50,23 @@ export function processPackageJSON(
     }
   }
 
+  // load exports if it's not the root pkg.json
+  if (content.exports && pkgRoot !== '/') {
```
Comment (Andarist) [inline packages/sandpack-core/src/resolver/utils/pkg-json.ts:54]: Note that likely you only need to still process alias fields - but main fields don't have to be processed.

Comment (DeMoorJasper) [reply packages/sandpack-core/src/resolver/utils/pkg-json.ts]: You're right good catch, updated it accordingly

**Review by CompuIves — APPROVED** [review]
Looks good!

**Comment by drcmda** [comment]
@CompuIves @Andarist @DeMoorJasper i think this one broke it once again. all our boxes are down, see: https://codesandbox.io/s/basic-demo-rrppl0y8l4

**Comment by DeMoorJasper** [comment]
> @CompuIves @Andarist @DeMoorJasper i think this one broke it once again. all our boxes are down, see: https://codesandbox.io/s/basic-demo-rrppl0y8l4 i don't know what the pr is trying to do, but if it would require we change something we wouldn't be able to survive it, we have too many sandboxes, poimandres with all its libraries is practically leaning on csb.

Sorry about this, will revert this PR for now and add some test cases to ensure this does not happen next time

The aim of this PR is to follow the exports spec better, supporting the new esm packages

**Comment by Andarist** [comment]
Do we know what broke this time? Which package was it and what could be the reason there?

**Comment by DeMoorJasper** [comment]
@Andarist yes, I've added some tests to this PR https://github.com/codesandbox/codesandbox-client/pull/7412 with the resolutions that broke

**Pull request merged**